third_party/move/tools/move-package-manifest/src/manifest.rs

```diff
@@ -0,0 +1,328 @@
+                    ))
+                },
+            },
+            _ => {
+                return Err(serde::de::Error::custom(
+                    "dependency cannot have have multiple locations",
+                ));
+            },
+        };
+
+        Ok(Dependency {
+            version: raw.version,
+            location,
+        })
+    }
+}
+
+/***************************************************************************************************
+ * Custom Serializer Implementations
+ *
+ **************************************************************************************************/
+impl Serialize for Version {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let s = format!("{}.{}.{}", self.major, self.minor, self.patch);
+        s.serialize(serializer)
+    }
+}
+
+impl Serialize for AddressOrWildcard {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            AddressOrWildcard::Wildcard => "_".serialize(serializer),
+            AddressOrWildcard::Numerical(addr) => addr.serialize(serializer),
+        }
+    }
+}
+
+impl Dependency {
+    fn into_raw(self) -> RawDependency {
+        let mut raw = RawDependency {
+            version: self.version,
+            ..Default::default()
+        };
+
+        match self.location {
+            PackageLocation::Local { path } => raw.local = Some(path),
+            PackageLocation::Git { url, rev, subdir } => {
+                raw.git = Some(url);
+                raw.rev = rev;
+                raw.subdir = subdir;
+            },
+            PackageLocation::Aptos {
+                node_url,
+                package_addr,
+            } => {
+                raw.aptos = Some(node_url);
+                raw.address = Some(package_addr);
+            },
+        }
+
+        raw
+    }
+
+    fn to_raw(&self) -> RawDependency {
+        self.clone().into_raw()
+    }
+}
+
+impl Serialize for Dependency {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.to_raw().serialize(serializer)
+    }
+}
+
+/***************************************************************************************************
+ * Default Values
+ *
+ **************************************************************************************************/
+#[allow(clippy::derivable_impls)]
+impl Default for Version {
+    fn default() -> Self {
+        Self {
+            major: 0,
+            minor: 0,
+            patch: 0,
+        }
+    }
+}
+
+impl Default for PackageMetadata {
+    fn default() -> Self {
+        Self {
+            name: PackageName::new("some_package").unwrap(),
+            version: Default::default(),
+            authors: BTreeSet::new(),
+            license: None,
+            upgrade_policy: None,
```

Comment:
Do we want packages to ever have `none` as the upgrade_policy? Is `none` interpreted differently than `compatible`? And if not, shouldn't we just use `compatible` here as default?

Reply:
`None` here allows us to express that the user has not specified an upgrade policy, which is subtly different from having compatible as a default. Being aware of this distinction may allow us to handle this better in later stages.